rboot-stage2a.c

```diff
@@ -13,42 +13,38 @@
 
 usercode* NOINLINE load_rom(uint32 readpos) {
 	
-	uint8 buffer[BUFFER_SIZE];
 	uint8 sectcount;
 	uint8 *writepos;
 	uint32 remaining;
 	usercode* usercode;
 	
-	rom_header *header = (rom_header*)buffer;
-	section_header *section = (section_header*)buffer;
+	rom_header header;
+	section_header section;
 	
 	// read rom header
-	SPIRead(readpos, header, sizeof(rom_header));
+	SPIRead(readpos, &header, sizeof(rom_header));
 	readpos += sizeof(rom_header);
 
 	// create function pointer for entry point
-	usercode = header->entry;
+	usercode = header.entry;
 	
 	// copy all the sections
-	for (sectcount = header->count; sectcount > 0; sectcount--) {
+	for (sectcount = header.count; sectcount > 0; sectcount--) {
 		
 		// read section header
-		SPIRead(readpos, section, sizeof(section_header));
+		SPIRead(readpos, &section, sizeof(section_header));
 		readpos += sizeof(section_header);
 
 		// get section address and length
-		writepos = section->address;
-		remaining = section->length;
+		writepos = section.address;
+		remaining = section.length;
 		
 		while (remaining > 0) {
 			// work out how much to read, up to 16 bytes at a time
-			uint32 readlen = (remaining < BUFFER_SIZE) ? remaining : BUFFER_SIZE;
+			uint32 readlen = (remaining < 0x1000) ? remaining : 0x1000;
```

Comment:
that's another one that's unlikely to change, but different purpose (and the reads here may not be sector aligned, so no obvious logic to connect the two) so I'd rather add a new one for this COPY_SIZE or something equally imaginative!